public/app/features/explore/TraceView/TraceView.tsx

```diff
@@ -16,15 +16,18 @@ import { useChildrenState } from './useChildrenState';
 import { useDetailState } from './useDetailState';
 import { useHoverIndentGuide } from './useHoverIndentGuide';
 import { colors, useTheme } from '@grafana/ui';
-import { TraceData, TraceSpanData, Trace, TraceSpan, TraceKeyValuePair, TraceLink } from '@grafana/data';
+import { TraceViewData, Trace, TraceSpan, TraceKeyValuePair, TraceLink } from '@grafana/data';
 import { createSpanLinkFactory } from './createSpanLink';
 
 type Props = {
-  trace: TraceData & { spans: TraceSpanData[] };
+  trace?: TraceViewData;
   splitOpenFn: (options: { datasourceUid: string; query: any }) => void;
 };
 
 export function TraceView(props: Props) {
+  if (!props.trace?.traceID) {
```

Comment:
my only concern with this is that it break the [rules of hooks](https://reactjs.org/docs/hooks-rules.html).

Reply:
Don't think so, the hooks are still at the toplevel inside that function. We just happen to execute something else before.

Reply:
I'm pretty sure it is breaking the rule by conditionally running the hooks but I'm okay if we merge it like this now.

Reply:
I am not 100% sure but I think this should be a problem if you run different number of hooks during render but if you run none that may be fine?

Reply:
I think it's all hooks (and in-order) or no hooks at all in a react view. At least that's what the link says.